interfaces/api/storage/IMasks2DManager.h

```diff
@@ -0,0 +1,120 @@
+namespace api {
+namespace storage {
+
+/**
+ * @class IMasks2DManager
+ * @brief <B>Allows to store a set of 2D masks.</B>
+ * <TT>UUID: 56f36544-85c4-4636-bd4e-b3493a9e95c4</TT>
+ *
+ * This storage component can be accessed by processing components to share persistent data.
+ */
+
+class XPCF_IGNORE IMasks2DManager : virtual public org::bcom::xpcf::IComponentIntrospect {
+public:
+    /// @brief IMasks2DManager default constructor
+    IMasks2DManager() = default;
+
+    /// @brief IMasks2DManager default destructor
+    virtual ~IMasks2DManager() = default;
+
+    /// @brief This method allows to add a mask to the 2D mask manager component
+    /// @param[in] mask the mask to add to the set of persistent masks
+    /// @param[in] defineMaskId if true an id will be set for the added mask, if false the id of the mask will be used
+    /// @return FrameworkReturnCode::_SUCCESS_ if the addition succeed, else FrameworkReturnCode::_ERROR.
+    virtual FrameworkReturnCode addMask(SRef<Mask2D> mask, bool defineMaskId = true) = 0;
+
+    /// @brief This method allows to get a mask by its id
+    /// @param[in] id id of the mask to get
+    /// @param[out] mask a mask stored in the masks manager
+    /// @return FrameworkReturnCode::_SUCCESS_ if succeed, else FrameworkReturnCode::_ERROR.
+    virtual FrameworkReturnCode getMask(uint32_t id, SRef<Mask2D>& mask) const = 0;
+
+    /// @brief This method allows to get a set of masks by their ids
+    /// @param[in] ids a vector of ids of the masks to get
+    /// @param[out] masks a vector of masks stored in the mask manager
+    /// @return FrameworkReturnCode::_SUCCESS_ if succeed, else FrameworkReturnCode::_ERROR.
+    virtual FrameworkReturnCode getMasks(const std::vector<uint32_t>& ids, std::vector<SRef<Mask2D>>& masks) const = 0;
+
+    /// @brief This method allows to get all masks
+    /// @param[out] masks the set of masks
+    /// @return FrameworkReturnCode::_SUCCESS_ if succeed, else FrameworkReturnCode::_ERROR.
+    virtual FrameworkReturnCode getAllMasks(std::vector<SRef<Mask2D>>& masks) const = 0;
+
+    /// @brief This method allows to know if a mask is already stored in the component
+    /// @param[in] id id of this mask
+    /// @return true if exist, else false
+    virtual bool isExistMask(uint32_t id) const = 0;
+
+    /// @brief This method allows to get the number of masks stored in the component
+    /// @return The number of masks
+    virtual size_t getNbMasks() const = 0;
+
+    /// @brief This method allow to suppress a mask by its id
+    /// @param[in] id id of the mask to suppress
+    /// @return FrameworkReturnCode::_SUCCESS_ if the suppression succeed, else FrameworkReturnCode::_ERROR.
+    virtual FrameworkReturnCode suppressMask(uint32_t id) = 0;
+
+    /// @brief This method is to set the mask collection
+    /// @param[in] maskCollection the mask collection of map
+    virtual void setMaskCollection(SRef<Mask2DCollection> maskCollection) = 0;
```

Comment:
Use 'const' for parameter:
`virtual void setMaskCollection(const SRef<Mask2DCollection> maskCollection) = 0;`